src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,688 @@
+                } else if (state.dataSurface->Surface(ig.SurfPtr).ExtBoundCond < 0 ||
+                           state.dataSurface->Surface(ig.SurfPtr).HeatTransferAlgorithm != DataSurfaces::HeatTransferModel::CTF) {
+                    ShowSevereError(state,
+                                    format("{}=\"{}\", invalid {} entered={}, not a valid surface for indoor green module",
+                                           RoutineName,
+                                           state.dataIPShortCut->cAlphaArgs(1),
+                                           state.dataIPShortCut->cAlphaFieldNames(2),
+                                           state.dataIPShortCut->cAlphaArgs(2)));
+                    ErrorsFound = true;
+                }
+            }
+            ig.SchedPtr = ScheduleManager::GetScheduleIndex(state, state.dataIPShortCut->cAlphaArgs(3));
+            if (ig.SchedPtr == 0) {
+                ShowSevereItemNotFound(state, eoh, state.dataIPShortCut->cAlphaFieldNames(3), state.dataIPShortCut->cAlphaArgs(3));
+                ErrorsFound = true;
+            } else { // check min/max on schedule
+                SchMin = ScheduleManager::GetScheduleMinValue(state, ig.SchedPtr);
+                SchMax = ScheduleManager::GetScheduleMaxValue(state, ig.SchedPtr);
+                if (SchMin < 0.0 || SchMax < 0.0) {
+                    if (SchMin < 0.0) {
+                        ShowSevereError(state,
+                                        format("{}=\"{}\", {}, minimum is < 0.0",
+                                               RoutineName,
+                                               state.dataIPShortCut->cAlphaArgs(1),
+                                               state.dataIPShortCut->cAlphaFieldNames(3)));
+                        ShowContinueError(
+                            state,
+                            format("Schedule=\"{}\". Minimum is [{:.1R}]. Values must be >= 0.0.", state.dataIPShortCut->cAlphaArgs(3), SchMin));
+                        ErrorsFound = true;
+                    }
+                    if (SchMax < 0.0) {
+                        ShowSevereError(state,
+                                        format("{}=\"{}\", {}, maximum is < 0.0",
+                                               RoutineName,
+                                               state.dataIPShortCut->cAlphaArgs(1),
+                                               state.dataIPShortCut->cAlphaFieldNames(3)));
+                        ShowContinueError(
+                            state,
+                            format("Schedule=\"{}\". Maximum is [{:.1R}]. Values must be >= 0.0.", state.dataIPShortCut->cAlphaArgs(3), SchMin));
+                        ErrorsFound = true;
+                    }
+                }
+            }
+            ig.etCalculationMethod = ETCalculationMethod::PenmanMonteith; // default
+            if (Util::SameString(Util::makeUPPER(state.dataIPShortCut->cAlphaArgs(4)), "PENMAN-MONTEITH")) {
```

Comment:
Do you know how to use the `getEnumValue` pattern? You add a `std::array<std::string_view, (int)ETCalculationMethod::Num> etCalculationMethodNamesUC` array to the `.hh` file. Then you can pass that and the input argument to `getEnumValue` and it will return the integer version of the `enum` that you can then `static_cast<ETCalculationMethod>`. There are many examples of this in the code.

Reply:
Thank you very much for the information! Updated. Using `getEnumValue` for enums in the getting the object input function.